app/services/processors/blog_technology_views_updater.rb

```diff
@@ -0,0 +1,41 @@
+module Processors
+  class BlogTechnologyViewsUpdater < BaseService
+    def call
+      blog_views_by_technology_and_month.each do |pair_group, visits|
+        timestamp, technology_id = pair_group
+        update_technology_visits_metric(technology_id, visits, timestamp)
+      end
+    end
+
+    private
+
+    def blog_views_by_technology_and_month
+      Metric.where(name: Metric.names[:blog_visits], ownable_type: BlogPost.to_s)
+            .where('value_timestamp >= ?', latest_technology_metrics_updated_at)
+            .joins('JOIN blog_posts on metrics.ownable_id = blog_posts.id')
```

Comment:
This was made like this because of the polymorphic association right?

Reply:
Yes! I had to write it that way because of that, I couldn't find any other way to do it.